src/extremeweatherbench/data_loader.py

```diff
@@ -0,0 +1,161 @@
+import fsspec
+import xarray as xr
+from extremeweatherbench import config, utils
+import pandas as pd
+import logging
+import dataclasses
+from typing import Tuple, Optional
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def open_forecast_dataset(
+    eval_config: config.Config,
+    forecast_schema_config: config.ForecastSchemaConfig,
+) -> xr.Dataset:
+    """Open the forecast dataset specified for evaluation.
+
+    Args:
+        eval_config: The evaluation configuration.
+        forecast_schema_config: The forecast schema configuration.
+
+    Returns:
+        The opened forecast dataset.
+    """
+    logger.debug("Opening forecast dataset")
+    if "://" in eval_config.forecast_dir:
```

Comment:
You probably need a note in the docstring providing information on how you infer the filesystem/location of the specified data., as well as the expected file formats (from below)